src/Microsoft.AspNetCore.SignalR.Common/Internal/Formatters/BinaryMessageParser.cs

```diff
@@ -6,69 +6,41 @@
 
 namespace Microsoft.AspNetCore.Sockets.Internal.Formatters
 {
-    public class BinaryMessageParser
+    public static class BinaryMessageParser
     {
-        private ParserState _state;
-
-        public void Reset()
+        public static bool TryParseMessage(ref ReadOnlyBuffer<byte> buffer, out ReadOnlyBuffer<byte> payload)
         {
-            _state = default(ParserState);
-        }
+            long length = 0;
+            payload = default(ReadOnlyBuffer<byte>);
 
-        public bool TryParseMessage(ref ReadOnlySpan<byte> buffer, out ReadOnlyBuffer<byte> payload)
-        {
-            if (_state.Length == null)
+            if (buffer.Length < sizeof(long))
             {
-                long length = 0;
-
-                if (buffer.Length < sizeof(long))
-                {
-                    payload = default(ReadOnlyBuffer<byte>);
-                    return false;
-                }
-
-                length = buffer.Slice(0, sizeof(long)).ReadBigEndian<long>();
-
-                if (length > Int32.MaxValue)
-                {
-                    throw new FormatException("Messages over 2GB in size are not supported");
-                }
-
-                buffer = buffer.Slice(sizeof(long));
-                _state.Length = (int)length;
+                return false;
```

Comment:
Shouldn't these methods basically throw in this case? At least at the moment we are reading the data and we are not returning it to the buffer so Try* does not make sense because we are discarding the data and hence corrupting messages received in multiple chunks.